### PR TITLE
Refector some image tests failing on arm simulator

### PR DESCRIPTION
**Pull request by yxlao**
Resolving some failing tests in #2343 in ARM simulator:
```
[  FAILED  ] 11 tests, listed below:
[  FAILED  ] Image.CreateFloatImage_3_2_Weighted
[  FAILED  ] Image.CreateFloatImage_3_2_Equal
[  FAILED  ] Image.CreateFloatImage_3_4_Weighted
[  FAILED  ] Image.CreateFloatImage_3_4_Equal
[  FAILED  ] RGBDImage.CreateFromColorAndDepth
[  FAILED  ] RGBDImage.CreateFromRedwoodFormat
[  FAILED  ] RGBDImage.CreateFromTUMFormat
[  FAILED  ] RGBDImage.CreateFromSUNFormat
[  FAILED  ] RGBDImage.CreateFromNYUFormat
[  FAILED  ] RGBDImage.FilterPyramid
[  FAILED  ] RGBDImage.CreatePyramid
```

The tests are failing because we're comparing float as 4 independent bytes. Since this PR is to support #2343, only a subset of the image tests are refactored.

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/2393)

### CODE REVIEW

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- cpp/tests/geometry/Image.cpp  27
         

Clones removed
==============
+ cpp/tests/geometry/RGBDImage.cpp  -30
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=6253362&bid=20361190)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- cpp/tests/geometry/Image.cpp  26
         

Clones removed
==============
+ cpp/tests/geometry/RGBDImage.cpp  -31
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=6253362&bid=20361190)

File: cpp/tests/geometry/RGBDImage.cpp [hunk 1]
```diff
@@ -70,396 +71,260 @@ TEST(RGBDImage, Constructor) {
...
+    float max_val;
+    std::tie(min_val, max_val) = FloatImageMinMax(im_rgbd->color_);
+    EXPECT_FLOAT_EQ(min_val, 0.0);
+    EXPECT_FLOAT_EQ(max_val, 1.0);
+    std::tie(min_val, max_val) = FloatImageMinMax(im_rgbd->depth_);
+    EXPECT_FLOAT_EQ(min_val, 0.0);
+    EXPECT_FLOAT_EQ(max_val, 6.889);
 }
 
 TEST(RGBDImage, CreateFromNYUFormat) {
```
Comment (ssheorey) [inline cpp/tests/geometry/RGBDImage.cpp:235]: We can use `GTEST_SKIP_("Reason for skipping")` in the body of the function.

**Review by ssheorey — COMMENTED** [review]
We've replaced a general test with multiple specific tests. Can you comment on the reason for the changes?

**Comment by yxlao** [comment]
> We've replaced a general test with multiple specific tests. Can you comment on the reason for the changes?
Which general test are you referring to? The total number of tests should be the same as before.

File: cpp/tests/geometry/RGBDImage.cpp (hunk 1 above)
Comment (yxlao) [reply cpp/tests/geometry/RGBDImage.cpp:235]: fixed

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- cpp/tests/geometry/Image.cpp  26
         

Clones removed
==============
+ cpp/tests/geometry/RGBDImage.cpp  -31
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=6253362&bid=20361190)

**Comment by ssheorey** [comment]
> > We've replaced a general test with multiple specific tests. Can you comment on the reason for the changes?
> 
> Which general test are you referring to? The total number of tests should be the same as before.

Earlier all the tests were parametrized calls to `TEST_CreateFloatImage()`, now each test is implemented separately.
Can you add some information on why the parametrized tests were failing on ARM / how this change fixes them?

**Comment by yxlao** [comment]
Yes. The main failing point is at `ExpectEQ(ref, float_image->data_);` where we used to compare float as 4 independent bytes. 

```cpp
void TEST_CreateFloatImage(
        const int& num_of_channels,
        const int& bytes_per_channel,
        const std::vector<uint8_t>& ref,
        const geometry::Image::ColorToIntensityConversionType& type) {
    ...
    ...
    ExpectEQ(ref, float_image->data_);  # <----- compare floats as bytes
}
```

Both `ref` and ` float_image->data_` are `std::vector<uint8_t>`, but in this particular tests, the image is float, so every 4 elements in `float_image->data_` is used to represent one float32. When comparing floats, we allow certain error margins. In this case, the ARM simulator's floating-point results are within the error margin. But if we treat the float as 4 bytes, the comparison doesn't allow any error margin, so we got errors complaining the result is off by one *bit*, e.g.

```
[ RUN      ] RGBDImage.CreateFromColorAndDepth
/home/runner/work/Open3D/Open3D/cpp/tests/test_utility/Compare.cpp:54: Failure
Expected equality of these values:
  v0[i]
    Which is: '8' (56, 0x38)
  v1[i]
    Which is: '7' (55, 0x37)
```

**Comment by ssheorey** [comment]
Thanks. I wanted to check if there was any way around the code duplication, like templates, but not sure if that would work with `googletest`. LGTM.

**Review by ssheorey — APPROVED** [review]

**Comment by yxlao** [comment]
thanks!

**Pull request merged**